Code/Framework/AzCore/AzCore/DOM/Backends/JSON/JsonSerializationUtils.h

```diff
@@ -0,0 +1,255 @@
+    //! be valid for the lifetime of buffer.
+    //! \param parseFlags (template) Settings for adjusting parser behavior.
+    //! \return The aggregate result specifying whether the visitor operations were successful.
+    template<ParseFlags parseFlags = ParseFlags::ParseComments>
+    Visitor::Result VisitSerializedJsonInPlace(AZStd::string& buffer, Visitor& visitor);
+
+    //! Takes a visitor specified by a callback and produces a rapidjson::Document.
+    //! \param writeCallback A callback specifying a visitor to accept to build the resulting document.
+    //! \return An outcome with either the rapidjson::Document or an error message.
+    AZ::Outcome<rapidjson::Document, AZStd::string> WriteToRapidJsonDocument(Backend::WriteCallback writeCallback);
+    //! Takes a visitor specified by a callback and reads them into a rapidjson::Value.
+    //! \param value The value to read into, its contents will be overridden.
+    //! \param allocator The allocator to use when performing rapidjson allocations (generally provded by the rapidjson::Document).
+    //! \param writeCallback A callback specifying a visitor to accept to build the resulting document.
+    //! \return An outcome with either the rapidjson::Document or an error message.
+    Visitor::Result WriteToRapidJsonValue(
+        rapidjson::Value& value, rapidjson::Value::AllocatorType& allocator, Backend::WriteCallback writeCallback);
+    //! Accepts a visitor with the contents of a rapidjson::Value.
+    //! \param value The rapidjson::Value to apply to visitor.
+    //! \param visitor The visitor to receive the contents of value.
+    //! \param lifetime The lifetime to specify for visiting strings. If the rapidjson::Value might be destroyed or changed
+    //! before the visitor is finished using these values, Lifetime::Temporary should be specified.
+    //! \return The aggregate result specifying whether the visitor operations were successful.
+    Visitor::Result VisitRapidJsonValue(const rapidjson::Value& value, Visitor& visitor, Lifetime lifetime);
+
+    template<ParseFlags parseFlags>
+    Visitor::Result VisitSerializedJson(AZStd::string_view buffer, Lifetime lifetime, Visitor& visitor)
+    {
+        rapidjson::Reader reader;
+        RapidJsonReadHandler handler(&visitor, lifetime);
+
+        // If the string is null terminated, we can use the faster AzStringStream path - otherwise we fall back on rapidjson::MemoryStream
+        if (buffer.data()[buffer.size()] == '\0')
+        {
+            AzStringStream stream(buffer);
+            reader.Parse<static_cast<int>(parseFlags)>(stream, handler);
+        }
+        else
+        {
+            rapidjson::MemoryStream stream(buffer.data(), buffer.size());
+            reader.Parse<static_cast<int>(parseFlags)>(stream, handler);
+        }
+        return handler.TakeOutcome();
+    }
+
+    template<ParseFlags parseFlags>
+    Visitor::Result VisitSerializedJsonInPlace(AZStd::string& buffer, Visitor& visitor)
```

Comment:
This function can actually accept a `char* buffer` and a `size_t size` argument and then that buffer can be passed to an `AzStringStream` for insitu parsing.

In that case I would rename `AzStringStream` to `AzInsituStream`